Add user-guide for custom permissions (#145)
rakutentech/android-miniapp: miniapp/USERGUIDE.md

        callback.invoke(true)
    }

    override fun requestCustomPermissions(
        permissions: List<Pair<MiniAppCustomPermissionType, String>>,

>         callback: (grantResult: String) -> Unit

Reviewer: Sorry, I missed this in the first PR. The callback in the public interface should be passed `List<Pair<MiniAppCustomPermissionType, MiniAppCustomPermissionResult>>` instead of a `String`. And then we can make `MiniAppCustomPermissionManager` be an internal class so the SDK can automatically handle serializing it to a JSON string for sending to the Mini App JS side.

I think this is how the iOS SDK is handling it ([see here](MiniAppCustomPermissionManager)). Is that right @ts-leojoseph-r ? iOS handles JSON serialization internally so the Host App just has to provide a specific model to the callback?

Also, the `ACCEPTED` permissions should be stored automatically by the SDK - so after the Host App provides the List of permissions to `callback.invoke`, the SDK should store them. @ts-leojoseph-r I think this is also how iOS behaves, right? The Host App doesn't need to manually call `MiniApp.setCustomPermissions` when using `MiniAppMessageBridge`?

Author: if HostApp doesn't need to manually call `MiniApp.setCustomPermissions`, then what is the reason to make it as public interface? 🤔

Reviewer: @khairul-alam-licon `MiniApp.setCustomPermissions` and `MiniApp.getCustomPermissions` are needed so the Host App can create a Mini App Permissions screen. This screen will allow the user to manually revoke permissions to each Mini App.

Reviewer 2: That's right @corycaywood . [MASDKCustomPermissionModel](https://github.com/rakutentech/ios-miniapp/blob/master/MiniApp/Classes/JavascriptBridge/MiniAppJavaScriptPermissionManager.swift#L140) conforms to Codable that helps the iOS SDK to serialize them into [JSON reponse](https://github.com/rakutentech/ios-miniapp/blob/master/MiniApp/Classes/JavascriptBridge/MiniAppJavaScriptPermissionManager.swift#L126).

Also, we are storing the ACCEPTED/DENIED permissions that user has responded [from the host app](https://github.com/rakutentech/ios-miniapp/blob/master/MiniApp/Classes/JavascriptBridge/MiniAppJavaScriptPermissionManager.swift#L68). Host app doesn't need to call the MiniApp.setCustomPermissions manually to store the permissions.

}
```

### #4 Custom Permissions

> MiniApp Android SDK supports list of Custom Permissions ( ```MiniAppCustomPermissionType```) and these can be stored and retrieved using the following public interfaces.

Reviewer: ```suggestion
MiniApp Android SDK supports list of Custom Permissions ( `MiniAppCustomPermissionType`) and these can be stored and retrieved using the following public interfaces.
```

#### Retrieving the Mini App Custom Permissions using MiniAppID

> Custom permissions and its status can be retrieved using the following interface. ```getCustomPermissions``` will return ```MiniAppCustomPermission``` that contains the meta-info such as ```Pair``` of 

Reviewer: ```suggestion
Custom permissions and its status can be retrieved using the following interface. `getCustomPermissions` will return `MiniAppCustomPermission` that contains the meta-info as a `Pair` of 
```

> name and grant result of the custom permissions per miniAppId.

Reviewer: ```suggestion
name and grant result (`ALLOWED` or `DENIED`). The custom permissions are stored per each miniAppId.
```

```kotlin
val permissions = miniapp.getCustomPermissions(miniAppId)
```
#### Store the Mini App Custom Permissions

> Custom permissions for a mini app is cached by the SDK and you can use the following interface to store and retrieve it when you need.

Reviewer: ```suggestion
Custom permissions for a mini app are cached by the SDK and you can use the following interface to store permissions when needed.
```
